internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -41,6 +41,9 @@ type Networkd struct {
 
 	hostname  string
 	resolvers []string
+
+	readyLock sync.Mutex
```

Comment:
We can also embed `sync.Mutex`, then we can just do `n.Lock()` and `n.Unlock`. That is a common convention I have along with calling this "`mu`", similar to the convention of using `wg` for `sync.WaitGroup`.